### PR TITLE
AUX Serial Throttling

### PR DESCRIPTION
**Pull request by zfields**

### CODE REVIEW

File: src/Notecard.cpp [hunk 1]
```diff
@@ -44,6 +44,14 @@
 
 #include "NoteTime.h"
 
+// AUX serial throttling is based on the Arduino define `SERIAL_RX_BUFFER_SIZE`.
+// Unfortunately, some platforms do NOT specify the define. In this case, 64
+// bytes is selected as the default value, because it is a common buffer size
+// across several platforms.
+#ifndef SERIAL_RX_BUFFER_SIZE
+#define SERIAL_RX_BUFFER_SIZE 64
```
Comment (zfields) [inline src/Notecard.cpp:52]: Should probably add a `#warning` to this, since this is not specified by the platform.

**Comment by zfields** [comment]
[Update `note-arduino` with AUX throttling](https://trello.com/c/XkaVMQE5/216-update-note-arduino-with-aux-throttling)

File: src/Notecard.cpp [hunk 2]
```diff
@@ -245,6 +253,15 @@ void Notecard::begin(NoteSerial * noteSerial_)
     if (noteSerial) {
         NoteSetFnSerial(noteSerialReset, noteSerialTransmit,
                         noteSerialAvailable, noteSerialReceive);
+
+        // Set the default debug serial throttling
+        J *req = NoteNewRequest("card.aux.serial");
+        if (req != NULL)
+        {
+            JAddIntToObject(req, "max", SERIAL_RX_BUFFER_SIZE - 1);
```
Comment (haydenroche5) [inline src/Notecard.cpp:261]: What's the -1 for?

File: test/mock/mock-arduino.hpp [hunk 3]
```diff
@@ -8,6 +8,8 @@
 #include <string>
 #include <vector>
 
+#define SERIAL_RX_BUFFER_SIZE 79
```
Comment (haydenroche5) [inline test/mock/mock-arduino.hpp:11]: Is 79 special in any way? Just curious.

**Review by haydenroche5 — APPROVED** [review]

File: test/mock/mock-arduino.hpp (hunk 3 above)
Comment (zfields) [reply test/mock/mock-arduino.hpp:11]: it's the year I was born. :smile:

File: src/Notecard.cpp (hunk 2 above)
Comment (zfields) [reply src/Notecard.cpp:261]: We observed that when the buffer is filled entirely, then it chokes on the last byte.

**Comment by m-mcgowan** [comment]
I missed the review before this was merged -  just noting that I've looked over the code. 👍

File: test/mock/mock-arduino.hpp (hunk 3 above)
Comment (m-mcgowan) [reply test/mock/mock-arduino.hpp:11]: Amazing that's the Chinese year of the taco.

Comment (zfields) [reply test/mock/mock-arduino.hpp:11]: lol. :100:

**Pull request merged**